course02/homework00/skeleton/vector.h

```diff
@@ -1,8 +1,45 @@
 #pragma once
 
+#include <array>
+
 class VectorInt
 {
-	// TODO
+	
+public:
+	int32_t size()
+	{
+		return mSize;
+	}
+
+	int32_t capacity()
+	{
+		return mData.size();
+	}
+
+	int32_t back()
+	{
+		if (mSize > 0)		
+			return mData[mSize-1];
+	}
+
+	void push_back(int32_t member)
+	{
+
+		if (mSize < mData.size())
+		{
+			mData[mSize] = member;
+			++mSize;
+		}
```

Comment:
I briefly mentioned this yesterday ; in most APIs you want to 1. be explicit 2. leave the choice to the user on what to do in unexpected scenario. Here, the unexpected scenario is to push an element in a full container. If you throw an exception, the user has the choice to catch it and do nothing, or go down, or try something else. Here, you don't give any choice to the user, because you make that choice yourself: if it's full, nothing happens. It's quite bad, as the user would have to check the size() of the contrainer before and after to be sure that the operation succeeded... !